### ISSUE TITLE
Keyboard sound plays with no keystroke input

### ISSUE THREAD

**Issue by MaisieOfMystery** [noblereign/GX-Mods-Extension#22]
Seemingly at random, Firefox will play a keyboard click noise despite no input from my keyboard. I figured it must have to be something with the keyboard itself, but the same issue persists even with a different keyboard. It even plays randomly when Firefox is minimized and I'm doing something in Discord/Spotify/etc.

It's not too common, and turning off the keyboard sounds at all fixes it, but just turning it off isn't exactly ideal. I've restarted my computer and uninstalled/reinstalled the extension, but the issue persists.

**Comment by noblereign** [comment]
do you have the "detect in address bar" option enabled? if so, it may be related to that, as it can cause false positives due to the admittedly scuffed methods required to make it work. 

if you _do_ have it on, then Firefox must be sending some requests that I haven't observed on my install, and I'll need to see if I can find them on a fresh install

if you _don't_ have it on, then something must be triggering the javascript key press event somehow... 🤔🤔

**Comment by MaisieOfMystery** [comment]
Yeah, I have "detect in address bar" turned off. This started happening after the update that implemented it, though, if that helps narrow it down.

**Comment by noblereign** [comment]
i have a feeling it might be this piece of code right here since it's the only time a keyboard sound would play outside of either actual keyboard presses or the address bar detection
![image](https://github.com/user-attachments/assets/94ed3ac0-2cf7-4bbc-8cfe-ec4fec93d349)
maybe `(cachedSettings.sfxKeyboard && cachedSettings.sfxAddressBar)` is returning true when it shouldn't be
i'll try changing the code and then releasing it in the next update, might take a bit though since i'm currently eating dinner

**Comment by MaisieOfMystery** [comment]
Nice! I will patiently await the update <3

**Comment by noblereign** [comment]
i just released 6.3.0 and i did try changing up the code a little, however i'm not sure if it fixes the issue. can you update and give it a try?

**Comment by MaisieOfMystery** [comment]
I've been using it all day and haven't encountered it once! That update seemed to do the trick, thank you so much ^^

**Comment by MaisieOfMystery** [comment]
Ack, false alarm. It's been happening again.

**Comment by noblereign** [comment]
mmm, i have no clue what it could be...
i think i might try creating a new version with a debug option to log when the keyboard sounds are played so we can figure out what's causing it, cause since you don't have the address bar option on i don't really know what could be causing false positives

**Comment by MaisieOfMystery** [comment]
It's ultimately super minor so not a big deal, but I will keep my eyes peeled if it's ever implemented <3 Thank you again either way, I super appreciate all this.

**Comment by noblereign** [comment]
i uploaded the version with the logging feature, version 6.3.1
you can access the debug options by clicking on the version number at the bottom:

![image](https://github.com/user-attachments/assets/e029697b-6ae6-439c-bac2-f9799d38965d)
![image](https://github.com/user-attachments/assets/41083919-00df-405a-8657-b6ec0f6f1efa)

whenever you feel like it, just toggle on the switch and do whatever until you notice it happen
once you notice it you should go back to the mod manager and export the log, it might be able to tell us more about why the sound is playing

**Comment by MaisieOfMystery** [comment]
Alright, I think I figured it out: all the keystrokes are coming from youtube tabs. Having fiddled with my youtube addons, it seems like it's conflicting with the "YouTube Redux" addon. No clue why it's doing that, but disabling that addon fixed the problem lol ^^

**Comment by noblereign** [comment]
looking at the code for youtube redux, it seems like it might be related to this:
![image](https://github.com/user-attachments/assets/587fd907-be83-423f-83a0-2e989aee78c7)
![image](https://github.com/user-attachments/assets/f1d19cd9-f8f0-4859-896f-6c709df88867)
there should be setting called "Auto-confirm 'video paused' popup", and the way that it seems to work is by faking a keyboard down event, causing GXM to pick it up and play a keyboard sound. if you turn that off you should be able to keep using Youtube Redux without having to deal with the random sound

**Comment by MaisieOfMystery** [comment]
Perfect! Thank you so much!! You're an angel ^^
